main/eval/src/mill/eval/EvaluatorCore.scala

```diff
@@ -169,7 +169,11 @@ private[mill] trait EvaluatorCore extends GroupEvaluator {
 
     EvaluatorCore.Results(
       goals.indexed.map(results(_).map(_._1).result),
-      finishedOptsMap.map(_._2).flatMap(_.toSeq.flatMap(_.newEvaluated)),
+      Strict.Agg.from(
+        Loose.Agg.from(
+          finishedOptsMap.values.flatMap(_.toSeq.flatMap(_.newEvaluated))
+        )
+      ),
       transitive,
```

Comment:
So as far as I can tell, `evaluated: Agg[Task[_]]` is only used in the test suite, and `transitive: Agg[Task[_]]` is not used at all. So this should be fine from a runtime behavior perspective. And anyway it's `private[mill]` so we change things without worrying about compatibility.

As for what the correct thing to do here is, Strict.Agg.from(Loose.Agg.from(` can just be a single `.distinct` call I think. Then maybe just a comment and a simple test case illustrating the edge case and then we should be good to go